parts/dcosmasterresources.t

```diff
@@ -141,9 +141,8 @@
         "name": "nicLoopNode"
       },
       "dependsOn": [
-{{if .MasterProfile.IsCustomVNET}}
         "[variables('masterNSGID')]",
-{{else}}
+{{if not .MasterProfile.IsCustomVNET}}        
```

Comment:
what is the purpose of listing variables in the dependsOn sections. It's not as if they will make a difference, right? If the variable isn't there, aren't you still going to get an upfront error when ARM parses the template?